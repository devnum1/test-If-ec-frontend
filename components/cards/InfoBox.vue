// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="info-box">
    <ul class="info-box__list">
      <li v-for="(item, index) in dataInfo" :key="index" class="info-box__item">
        <div class="info-box__title text-">
          {{ item.title }}
        </div>
        <div class="info-box__text text-right text-darker">
          <span>{{ item.text }}</span
          ><span v-if="item.unit" class="unit">{{ item.unit }}</span>
        </div>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    dataInfo: {
      type: [Array, Object],
      default: () => [{}],
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/components/info-box';
</style>
