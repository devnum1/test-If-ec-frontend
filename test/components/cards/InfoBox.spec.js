import { shallowMount } from '@vue/test-utils';
import InfoBox from '@/components/cards/InfoBox.vue';

describe('InfoBox', () => {
  const petInfo = [
    {
      title: '誕生日',
      text: '1990/07/25',
    },
    {
      title: '性別',
      text: '女',
    },
    {
      title: '郵便番号',
      text: '150-0046',
    },
    {
      title: '電話番号',
      text: '08012345678',
    },
    {
      title: '連絡用メールアドレス',
      text: 'y.hanako12@peco-japan.com',
    },
    {
      title: 'PECOからのお知らせ',
      text: '受け取る',
    },
  ];

  const flushPromises = () => new Promise(setImmediate);
  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(InfoBox, {
      propsData: {
        dataInfo: petInfo,
      },
    });
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    const wrapper = shallowMount(InfoBox);
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  // for each item display key - value
  test('List.vue to display list', async () => {
    await flushPromises();

    expect(wrapper.find('.info-box__item').exists()).toBe(true);

    const items = wrapper.findAll('.info-box__item');
    items.wrappers.forEach((wrapper, i) => {
      let title = wrapper.find('.info-box__title');
      expect(title.exists()).toBe(true);
      expect(title.text()).toBe(petInfo[i].title);

      let infotext = wrapper.find('.info-box__text');
      expect(infotext.exists()).toBe(true);
      expect(infotext.text()).toBe(petInfo[i].text);
    });
  });
});
