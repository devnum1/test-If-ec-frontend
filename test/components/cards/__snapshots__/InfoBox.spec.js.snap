// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox match Snapshot 1`] = `
"<div class=\\"info-box\\">
  <ul class=\\"info-box__list\\">
    <li class=\\"info-box__item\\">
      <div class=\\"info-box__title text-\\">
        誕生日
      </div>
      <div class=\\"info-box__text text-right text-darker\\"><span>1990/07/25</span>
        <!---->
      </div>
    </li>
    <li class=\\"info-box__item\\">
      <div class=\\"info-box__title text-\\">
        性別
      </div>
      <div class=\\"info-box__text text-right text-darker\\"><span>女</span>
        <!---->
      </div>
    </li>
    <li class=\\"info-box__item\\">
      <div class=\\"info-box__title text-\\">
        郵便番号
      </div>
      <div class=\\"info-box__text text-right text-darker\\"><span>150-0046</span>
        <!---->
      </div>
    </li>
    <li class=\\"info-box__item\\">
      <div class=\\"info-box__title text-\\">
        電話番号
      </div>
      <div class=\\"info-box__text text-right text-darker\\"><span>08012345678</span>
        <!---->
      </div>
    </li>
    <li class=\\"info-box__item\\">
      <div class=\\"info-box__title text-\\">
        連絡用メールアドレス
      </div>
      <div class=\\"info-box__text text-right text-darker\\"><span>y.hanako12@peco-japan.com</span>
        <!---->
      </div>
    </li>
    <li class=\\"info-box__item\\">
      <div class=\\"info-box__title text-\\">
        PECOからのお知らせ
      </div>
      <div class=\\"info-box__text text-right text-darker\\"><span>受け取る</span>
        <!---->
      </div>
    </li>
  </ul>
</div>"
`;
